### DISCUSSION TITLE
Instantaneous active power imported from grid?

### DISCUSSION THREAD

**Discussion by Deck4847** [lbbrhzn/ocpp#505] [Q&A]
I just recently installed my Wallbox Pulsar Plus and I started using this OCPP integration and it's been great so far.

Today I just installed and setup power boost using a Temco SPM1-100-AC clamp. Power boost is active in the Wallbox app and in OCPP I can select an option to record the active energy imported from grid, but would it be possible to have an entity that displays the instantaneous active power from grid also? Or is that sensor data not exposed by the charger?

Right now I can see the instantaneous active power imported by the EV, but being able to see the power from grid to the whole house also via the energy monitor clamp would be very useful.

**Comment by GrantKeymer** [comment]
> ...would it be possible to have an entity that displays the instantaneous active power from grid also? Or is that sensor data not exposed by the charger?

Unfortunately it doesn't look as though the Pulsar Plus provides any metric for that, hence it isn't possible to display in HA with the current version of charger firmware.  Maybe something Wallbox could provide in future?

For myself, I've given up trying to use an Energy Meter connected to the Pulsar Plus and have a [Shelly EM](https://shelly.cloud/products/shelly-em-smart-home-automation-device/) on order, due to arrive any day now.  It will allow me to see power imported from the grid, or power exported to the grid from my solar panels.  Shelly devices are supported with an HA integration, so it will be easy to automate charging however I wish.

Shelly EM is quite cheap at only 41.90€ so it would solve your problem nicely.

**Reply by Deck4847** [reply]
Ok, that's unfortunate. I actually already use an energy monitor that I built using an ESP32 connected to HA via ESPhome. I was just hoping that the wallbox sensor could take over and I could use my ESP to measure something else.
